lib/src/Uncertainty/Algorithm/MetaModel/Kriging/GaussianProcessFitter.cxx

```diff
@@ -0,0 +1,738 @@
+}
+
+GaussianProcessFitter::GaussianProcessFitter(const Sample & inputSample,
+    const Sample & outputSample,
+    const CovarianceModel & covarianceModel,
+    const Basis & basis)
+  : MetaModelAlgorithm(inputSample, outputSample)
+  , covarianceModel_()
+  , reducedCovarianceModel_()
+  , solver_()
+  , optimizationBounds_()
+  , beta_(0)
+  , rho_(0)
+  , F_(0, 0)
+  , result_()
+  , basis_()
+  , covarianceCholeskyFactor_()
+  , covarianceCholeskyFactorHMatrix_()
+  , keepCholeskyFactor_(ResourceMap::GetAsBool("GaussianProcessFitter-KeepCovariance"))
+  , optimizeParameters_(ResourceMap::GetAsBool("GaussianProcessFitter-OptimizeParameters"))
+  , analyticalAmplitude_()
+  , lastReducedLogLikelihood_(SpecFunc::LowestScalar)
+{
+  // Set covariance model
+  setCovarianceModel(covarianceModel);
+
+  if (basis.getSize() > 0)
+    setBasis(basis);
+
+  initializeMethod();
+  initializeDefaultOptimizationAlgorithm();
+}
+
+/* Covariance model accessors */
+void GaussianProcessFitter::setCovarianceModel(const CovarianceModel & covarianceModel)
+{
+  // Here we can store any modified version of the given covariance model wrt its parameters as it is mainly a parametric template
+  const UnsignedInteger inputDimension = inputSample_.getDimension();
+  const UnsignedInteger dimension = outputSample_.getDimension();
+
+  if (covarianceModel.getInputDimension() != inputDimension)
+    throw InvalidArgumentException(HERE) << "Covariance model input dimension is " << covarianceModel.getInputDimension() << ", expected " << inputDimension;
+  if (covarianceModel.getOutputDimension() != dimension)
+    throw InvalidArgumentException(HERE) << "Covariance model output dimension is " << covarianceModel.getOutputDimension() << ", expected " << dimension;
+  covarianceModel_ = covarianceModel;
+  // All the computation will be done on the reduced covariance model. We keep the initial covariance model (ie the one we just built) in order to reinitialize the reduced covariance model if some flags are changed after the creation of the algorithm.
+  reducedCovarianceModel_ = covarianceModel_;
+  // Now, adapt the model parameters.
+  // First, check if the parameters have to be optimized. If not, remove all the active parameters.
+  analyticalAmplitude_ = false;
+  if (!optimizeParameters_) reducedCovarianceModel_.setActiveParameter(Indices());
+  // Second, check if the amplitude parameter is unique and active
+  else if (ResourceMap::GetAsBool("GaussianProcessFitter-UseAnalyticalAmplitudeEstimate"))
+  {
+    // The model has to be of dimension 1
+    if (reducedCovarianceModel_.getOutputDimension() == 1)
+    {
+      const Description activeParametersDescription(reducedCovarianceModel_.getParameterDescription());
+      // And one of the active parameters must be called amplitude_0
+      for (UnsignedInteger i = 0; i < activeParametersDescription.getSize(); ++i)
+        if (activeParametersDescription[i] == "amplitude_0")
```

Comment:
What if the user renames that parameter?

Reply:
@mbaudin47 The user cannot change the parameters description. There is no method exposed to do that, on purpose: we want to rely on the parameters names to know what is going on in the fitting process